lumigator/python/mzai/sdk/experiments.py

```diff
@@ -0,0 +1,77 @@
+from uuid import UUID
+from http import HTTPMethod
+from json import dumps
+
+from schemas.experiments import (
+    ExperimentCreate,
+    ExperimentResponse,
+    ExperimentResultDownloadResponse,
+    ExperimentResultResponse,
+)
+from schemas.extras import ListingResponse
+
+from client import ApiClient
+
+
+class Experiments:
+    EXPERIMENTS_ROUTE = "experiments"
+
+    def __init__(self, c: ApiClient):
+        self.__client = c
+
+    def create_experiment(self, experiment: ExperimentCreate) -> ExperimentResponse:
+        """Creates a new experiment."""
+        response = self.__client.get_response(
+            self.EXPERIMENTS_ROUTE, HTTPMethod.POST, dumps(experiment)
+        )
+
+        if not response:
+            return []
+
+        data = response.json()
+        return ExperimentResponse(**data)
+
+    def get_experiment(self, experiment_id: UUID) -> ExperimentResponse:
+        """Returns information on the experiment for the specified ID."""
+        response = self.__client.get_response(f"{self.EXPERIMENTS_ROUTE}/{experiment_id}")
+
+        if not response:
+            return []
```

Comment:
might be worth also adding something additional to indicate to folks whether the issue is at the API level, the client level, etc https://requests.readthedocs.io/en/latest/user/quickstart/#response-status-codes

Reply:
Thanks @veekaybee yep, this is something we probably want to consider more widely across the SDK as a way to convey problems to the consumer. I've raised a ticket for this internally to decide how we want to address it. 😄